### PR TITLE
Sam Phase 4: Redirect all managed group endpoints to talk to Sam instead of Rawls

### PR DESCRIPTION
**Pull request by MatthewBemis**
https://broadinstitute.atlassian.net/browse/GAWB-2532

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/UserApiServiceSpec.scala [hunk 1]
```diff
@@ -281,69 +232,6 @@ class UserApiServiceSpec extends BaseServiceSpec with RegisterApiService with Us
       }
     }
 
-    "when GET-ting my group membership" - {
```
Comment (marctalbott) [inline src/test/scala/org/broadinstitute/dsde/firecloud/service/UserApiServiceSpec.scala:284]: What happened to these tests? Are they unnecessary now or were you planning on moving them somewhere? Maybe a `ManagedGroupServiceSpec`?

**Review by marctalbott — APPROVED** [review]

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/UserApiServiceSpec.scala (hunk 1 above)
Comment (MatthewBemis) [reply src/test/scala/org/broadinstitute/dsde/firecloud/service/UserApiServiceSpec.scala:284]: good catch. i copied almost all of these changes from an old branch of mine here which has the replacement tests i should have carried over: https://github.com/broadinstitute/firecloud-orchestration/compare/mb-sam-groups#diff-20848548f8fd1b7fb280ce41a8c273faR1

File: src/main/scala/org/broadinstitute/dsde/firecloud/service/RegisterService.scala [hunk 2]
```diff
@@ -67,7 +67,6 @@ class RegisterService(val rawlsDao: RawlsDAO, val samDao: SamDAO, val thurloeDao
   private def registerUser(userInfo: UserInfo): Future[RegistrationInfo] = {
     for {
       registrationInfo <- samDao.registerUser(userInfo)
-      _ <- rawlsDao.registerUser(userInfo) //This call to rawls handles leftover registration pieces (welcome email and pending workspace access)
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/firecloud/service/RegisterService.scala:74]: where did welcome email go?

**Review by dvoet — APPROVED** [review]

File: src/main/scala/org/broadinstitute/dsde/firecloud/service/NihService.scala [hunk 3]
```diff
@@ -130,7 +129,7 @@ trait NihService extends LazyLogging {
 
       // The request to rawls to completely overwrite the group
```
Comment (ahaessly) [inline src/main/scala/org/broadinstitute/dsde/firecloud/service/NihService.scala:142]: update comment to sam instead of rawls

**Review by ahaessly — APPROVED** [review]
looks good. i will need to create an all_broad_users managed group before this code is merged to dev. the ticket for that work is GAWB-3854

**Comment by dvoet** [comment]
jenkins retest - sbt problem

File: src/main/scala/org/broadinstitute/dsde/firecloud/FireCloudConfig.scala [hunk 4]
```diff
@@ -193,4 +193,9 @@ object FireCloudConfig {
     val entityWorkspaceName: Option[String] = if (metrics.hasPath("entityWorkspaceName")) Some(metrics.getString("entityWorkspaceName")) else None
     val libraryNamespaces: List[String] = metrics.getStringList("libraryWorkspaceNamespace").asScala.toList
   }
+
+  object Notification {
+    private val notification = config.getConfig("notification")
```
Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/firecloud/FireCloudConfig.scala:207]: i can't find the fc-develop branch with the config change here, and it's causing orch to not come up in my phase4 fiab. any clues? @dvoet @MatthewBemis

**Pull request merged**